.github/workflows/release.yml

```diff
@@ -31,12 +31,23 @@ jobs:
     name: Release and Publish
     runs-on: ubuntu-latest
     needs: test
+    env:
+      VSCE_TOKEN: ${{secrets.VSCE_TOKEN}}
     steps:
       - uses: actions/checkout@master
+
       - uses: actions/setup-node@v1
         with:
           node-version: ^12.13
-      - name: make.js - build, test, and package
-        run: |
-          yarn install
-          node make.js all
+
+      - name: Yarn install
+        run: yarn install
+
+      - name: Build, lint, test and package
+        # Stuck on v1.0  because of https://github.com/GabrielBB/xvfb-action/issues/10
+        uses: GabrielBB/xvfb-action@v1.0
+        with:
+          run: node make.js all
+
+      - name: Publish to VSCode extensions store
+        run: node make.js publish -- ${{env.VSCE_TOKEN}}
```

Comment:
@philsturgeon This will attempt at publishing a new version on each merge. This will only succeed when the package.json version has just been bumped and no similar version has already been published. Otherwise it will err (cf. https://github.com/Microsoft/vscode-vsce/issues/156).

As such, merging this PR:
- should try and publish a new version
- should fail if you already have manually published v0.2.1
- should succeed and publish v0.2.1 if you haven't

Reply:
@philsturgeon another option would be to "hide" this publishing step only when a new 'v*' tag is pushed (cf. https://github.community/t/how-to-run-github-actions-workflow-only-for-new-tags/16075/10)

Thoughts?

Reply:
@philsturgeon After some hours of sleep, I think I prefer the second option for now: publishing on tag pushing. Thoughts?

Reply:
We could create a lot of this logic but we were initially using semantic-release which is a convention based npm module that handles a lot of it for us. This was removed in the merge, but had a lovey plugin (might have been this one, there’s a few https://github.com/raix/semantic-release-vsce) that would handle a lot of the publishing with the token being an env variable.

What I liked about this was that there was no make.js which meant no logic on our side, no figuring out version numbers, no worrying about how to publish or when, and it also meant it worked the same as other tools for which I’ve aimed to simplify releases https://github.com/openapi-contrib/openapi-schema-to-json-schema

Maybe we could get back to that, instead of this more homegrown solution?

Reply:
@philsturgeon Few thoughts:
- semantic versioning requires some love from the commit messages
- make.js does a lot of thing and `semantic-release-vsce` very little
- deploying is currently done manually (by you) and is a pain

Although we should rely on semantic version, getting there in one shot may not be that easy

Proposal is to get there through small steps:
- ensure commit messages can be parsed (#14)
- use semantic versioning to automate publishing to github releases (in a later PR)
- use semantic versioning to automate publishing to VSCode marketplace (in a later PR)

And meanwhile rely on tag pushing to do the automated publishing to VSCode marketplace.

Thoughts?

Reply:
I'm not following. 

> make.js does a lot of thing and semantic-release-vsce very little

What specifically? We had a task which would "compile" things, testing was very similar, then the publish was done by semantic-release-vsce instead of the make.js publish task. 

> deploying is currently done manually (by you) and is a pain

It wasn't! :D GitHub Releases and VS Code Marketplace publishing was done with two different semantic-release plugins:

- `"@semantic-release/github": "^7.0.5",`
- `"semantic-release-vsce": "^3.0.1",`

The config that made that work was in `package.json`

```json
  "release": {
    "branches": "master",
    "verifyConditions": [
      "semantic-release-vsce",
      "@semantic-release/github"
    ],
    "prepare": {
      "path": "semantic-release-vsce",
      "packageVsix": "stoplight-spectral.vsix"
    },
    "publish": [
      "semantic-release-vsce",
      {
        "path": "@semantic-release/github",
        "assets": "stoplight-spectral.vsix"
      }
    ]
  }
```

That specific config can be found here: https://github.com/stoplightio/vscode-spectral/commit/5ab41cbd6a38c569ff78a1a62a138809a3a26171#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L130


Travis removed this stuff and we said we'd bring it back, but it's hard to do without understanding more of the current situation. 

Maybe digging through that commit to see what semantic release related things have been removed is handy? 

We never bothered enforcing the commit messages, as that just annoys contributors. We just have to remember to give it a good name on merge commit and its all fine.

Reply:
@philsturgeon As discussed, I've created #15 to keep track of the next steps.

Reply:
Great! Let’s do things the way you’re saying and get done with that later. Merge away! 🥳